### PR TITLE
fix: [ANDROAPP-3157] Dataset design review

### PR DESCRIPTION
**Pull request by Balcan**
## Description
[ jira issue ](https://jira.dhis2.org/browse/ANDROAPP-3157)
BREAKING CHANGE This issue has changed the default date format for UI. We should check everywhere in the app where the ui date is shown to be sure it is behaving as expected.
## Covered unit test cases
Describe the tests that you ran to verify your changes.
## Where did you test this issue?
- [ ] Smartphone Emulator
- [ ] Tablet Emulator
- [x] Smartphone
- [x] Tablet
## Which Android versions did you test this issue?
- [ ] 4.4
- [ ] 5.X - 6.X
- [x] 7.X
- [x] 8.X
- [ ] 9.X - 10.X
- [ ] Other
## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code

### CODE REVIEW

File: app/src/androidTest/java/org/dhis2/usescases/teidashboard/TeiDashboardTest.kt [hunk 1]
```diff
@@ -254,8 +254,10 @@ class TeiDashboardTest : BaseTest() {
         }
 
         eventRobot {
+            waitToDebounce(10000)
             clickOnChangeDate()
             clickOnEditDate()
+            waitToDebounce(10000)
```
Comment (JaimeToca) [inline app/src/androidTest/java/org/dhis2/usescases/teidashboard/TeiDashboardTest.kt:260]: you don't need this two waitToDebounce

File: app/src/main/java/org/dhis2/usescases/datasets/dataSetTable/dataSetDetail/DataSetDetailFragment.kt [hunk 2]
```diff
@@ -119,11 +119,11 @@ class DataSetDetailFragment private constructor() : FragmentGlobalAbstract(), Da
             if (dataSetInstance.completed()) {
                 completedDate.visibility = View.VISIBLE
                 completedDate.text = String.format(
-                    "%s %s",
-                    getString(R.string.data_set_closed),
+                    "Completed by %s: %s",
```
Comment (JaimeToca) [inline app/src/main/java/org/dhis2/usescases/datasets/dataSetTable/dataSetDetail/DataSetDetailFragment.kt:122]: no translation?

File: app/src/test/java/org/dhis2/utils/DateUtilsTest.java [hunk 3]
```diff
@@ -68,303 +69,303 @@ public void expiryPeriodAndDaysInRange() throws ParseException {
...
+        assertEquals("2017-07-01", DateUtils.oldUiDateFormat().format(minDateSixMonthly2));
+        assertEquals("2018-04-01", DateUtils.oldUiDateFormat().format(minDateSixMonthlyApril));
+        assertEquals("2017-10-01", DateUtils.oldUiDateFormat().format(minDateSixMonthlyApril2));
+        assertEquals("2018-01-01", DateUtils.oldUiDateFormat().format(minDateYearly));
+        assertEquals("2017-01-01", DateUtils.oldUiDateFormat().format(minDateYearly2));
+        assertEquals("2018-05-01", DateUtils.oldUiDateFormat().format(minDateFinancialApril));
+        assertEquals("2017-05-01", DateUtils.oldUiDateFormat().format(minDateFinancialApril2));
+        assertEquals("2018-07-01", DateUtils.oldUiDateFormat().format(minDateFinancialJuly));
+        assertEquals("2017-07-01", DateUtils.oldUiDateFormat().format(minDateFinancialJuly2));
+        assertEquals("2017-10-01", DateUtils.oldUiDateFormat().format(minDateFinancialOct));
```
Comment (JaimeToca) [inline app/src/test/java/org/dhis2/utils/DateUtilsTest.java:97]: why asserting the oldUIDateFormat, and not the new one?

File: app/src/androidTest/java/org/dhis2/usescases/teidashboard/robot/EnrollmentRobot.kt [hunk 4]
```diff
@@ -39,8 +40,9 @@ class EnrollmentRobot : BaseRobot() {
 
     fun clickOnPersonAttributes(attribute: String) {
         onView(withId(R.id.fieldRecycler))
-            .perform(actionOnItem<EditTextCustomHolder>(
-                hasDescendant(withText(attribute)), click()))
+            .perform(
+                scrollToBottomRecyclerView(),
+                actionOnItemAtPosition<RecyclerView.ViewHolder>(5, click()))
```
Comment (JaimeToca) [inline app/src/androidTest/java/org/dhis2/usescases/teidashboard/robot/EnrollmentRobot.kt:45]: why fixed position 5? This test might actually fail depending on the device screen.

File: app/src/main/java/org/dhis2/usescases/datasets/dataSetTable/DataSetTableActivity.java [hunk 5]
```diff
@@ -365,4 +402,41 @@ private void configureShapeDrawable() {
...
+                analyticsHelper().setEvent(SHOW_HELP, CLICK, SHOW_HELP);
+                showTutorial(true);
+            } else if (itemId == R.id.reopen) {
+                presenter.reopenDataSet();
+            }
+            return true;
+
+        });
+        popupMenu.show();
+    }
```
Comment (JaimeToca) [inline app/src/main/java/org/dhis2/usescases/datasets/dataSetTable/DataSetTableActivity.java:441]: This can be a separate component maybe?

**Comment by JaimeToca** [comment]
Hello @Balcan , good job !! :medal_military: . Added some comments

File: app/src/test/java/org/dhis2/utils/DateUtilsTest.java (hunk 3 above)
Comment (Balcan) [reply app/src/test/java/org/dhis2/utils/DateUtilsTest.java:97]: just checking the result

File: app/src/main/java/org/dhis2/usescases/datasets/dataSetTable/DataSetTableActivity.java [hunk 6]
```diff
@@ -365,4 +402,82 @@ private void configureShapeDrawable() {
...
+                    return true;
+                })
+                .build()
+                .show();
+    }
+
+    private void showReopenDialog() {
+        AlertBottomDialog.Companion.getInstance()
+                .setTitle("Are you sure?")
+                .setMessage("Do you want to re-open the data set?")
```
Comment (JaimeToca) [inline app/src/main/java/org/dhis2/usescases/datasets/dataSetTable/DataSetTableActivity.java:443]: is translation needed here?

Comment (Balcan) [reply app/src/main/java/org/dhis2/usescases/datasets/dataSetTable/DataSetTableActivity.java]: done!

**Pull request merged**